### DISCUSSION TITLE
WHy is it so slow? (2)

### DISCUSSION THREAD

**Discussion by babakfp** [huntabyte/shadcn-svelte#1112] [General]
Hi

A few days ago, I created a new SvelteKit project and Just added all the shadcn-svelte components in. Added a few components (by using the code from examples) to a page, and every save took many seconds to update the page.

What is causing this issue? Is this a Svelte problem or SvelteKit or Vite, or what?

**Comment by seb-goodman** [comment]
I'm guessing there was some dependency optimisation happening in the background. This will only happen after new dependencies are installed and won't affect deployed apps.

**Comment by Leonhart130** [comment]
Why so many depedency optimisation ? There is definitely something wrong here, I tried several svelte libraries, and no one caused so many slow and frequent 'depedency optimization' reloads, sometimes not even touching the component might still trigger you an 'optimized depedency' message with a 15 seconds refresh at least

**Comment by seb-goodman** [comment]
Give something like this a go in the vite.config.ts:

```
import { sveltekit } from '@sveltejs/kit/vite';
import { defineConfig } from 'vite';

export default defineConfig({
	plugins: [sveltekit()],
	optimizeDeps: {
		entries: ['./src/routes/**/*']
	},
	server: {
		warmup: {
			clientFiles: ['./src/routes/**/*']
		}
	}
});
```

Alternatively I suppose you can run the server via ```run build/preview``` rather than dev, which is what I assume you're currently using.
